xarray/core/dataset.py

```diff
@@ -2416,6 +2416,60 @@ def from_dataframe(cls, dataframe):
             obj[name] = (dims, data)
         return obj
 
+    def to_dask_dataframe(self, set_index=False):
+        """
+        Convert this dataset into a dask.dataframe.DataFrame.
+
+        Both the data and coordinate variables in this dataset form the columns of the
+        DataFrame.
+
+        If set_index=True, the dask DataFrame is indexed by
+        this dataset's coordinate. Since dask DataFrames to not support
+        multi-indexes, set_index only works if there is one coordinate dimension.
+        """
+
+        import dask.dataframe as dd
+
+        ordered_dims = self.dims
+        coord_columns = [c for c in self if c in ordered_dims]
+
+        # convert all coordinates into data variables
+        ds = self.reset_index(coord_columns).reset_coords()
+
+        # ensure we are dealing with dask arrays with consistent chunks
+        chunked = ds.chunk(chunks=self.chunks)
+
+        # order columns so that coordinates appear before data
+        columns = list(chunked)
+        num_coords = len(coord_columns)
+        columns = columns[-num_coords:] + columns[:-num_coords]
```

Comment:
This logic is a little fragile -- it relies on the iteration order of xarray datasets, which although unlikely to change is not really guaranteed to have coordinates last.

It would be better to build this up explicitly from the dataset, e.g.,
```
columns = list(ds.dims)
columns.extend(k for k in ds.coords if k not in ds.dims)
columns.extend(ds.data_vars)
```

Reply:
Ah! that is much better.  I'll do what I think is the simplest thing:

```
columns = list(self.coords) + list(self.data_vars)
```